src/runtime/backend.cpp

```diff
@@ -73,6 +74,15 @@ backend_manager::backend_manager()
       }
     }
   });
+
+  if(std::none_of(_backends.cbegin(), _backends.cend(), 
+                  [](const std::unique_ptr<backend>& b){
+                    return b->get_hardware_platform() == hardware_platform::cpu;
+                    }))
+  {
+    HIPSYCL_DEBUG_ERROR << "No CPU backend has been loaded. Things will break." << std::endl;
+    throw sycl::runtime_error{"No CPU backend has been loaded."};
```

Comment:
We should `std::terminate()` imo. This is not a situation where we want the user to have the option to catch the exception and continue as the entire setup is broken at this point.

Reply:
This always assumes that SYCL is the only reason for the application to run... what if an application just tries to load hipSYCL and if something fails, would back off to another backend at runtime..?
I know that is pretty hypothetical and might get messy, but I'm not really a fan for a _library_ to decide about whether the actual application should terminate...

Reply:
Except it's not just a library, it's also a compiler stack that is used to compile the application. Since you cannot compile code with two compilers, I actually think a SYCL implementation is well within its rights to assume special privileges :)

It's not uncommon for HPC stacks to terminate in case of problems that make recovering impossible, e.g. MPI implementations also do this.

If you have borked your setup so much that the hipSYCL installation is broken there's something fundamentally wrong with your build environment.